org.eclipse.jdt.debug/model/org/eclipse/jdt/debug/core/IJavaStackFrame.java

```diff
@@ -35,6 +35,43 @@
 public interface IJavaStackFrame extends IStackFrame, IJavaModifiers,
 		IFilteredStep, IDropToFrame {
 
+	/**
+	 * Categorization of the stack frame.
+	 *
+	 * @since 3.22
+	 *
+	 */
+	enum Category {
```

Comment:
It feels wrong to publish this enum as API as it may be subject to change. Especially "platform" may be usefull but has no clear definition. Also those categories may not be disjunct. code could be synthetic platform test and library at once. One could either use plain Strings as Category or have a CategoryRegistry where new Categories could be added at runtime.

Reply:
Yes, the categories are not exact and there could be frames that could belong to multiple categories.
However, I don't think that adding new categories is a feasible - and useful feature ever. Main problem is that a single category needs to be handled in multiple places and going for a very general solution would be hard. For example, to have a foreground and background colors we have to declare them in the plugin.xml, otherwise it can't be set in the color editing preference page.
Maybe, we could have Category as a 'record' and have them declared as a fixed constant in the JDIDebugPlugin.

Reply:
As long as the solution does not make the categories public API, it may be ok.

Reply:
What do you think about this solution?